### PR TITLE
Readers for SpRL examples

### PR DESCRIPTION
**Pull request by Rahgooy**

### CODE REVIEW

**Comment by kordjamshidi** [comment]
Thank you @Rahgooy for preparing the readers for an SpRL example.

**Comment by Rahgooy** [comment]
You are welcome

**Comment by danyaljj** [comment]
@Rahgooy The files have formatting problems;  could you run  `compile` and `format` within `sbt` and push the changes?

**Comment by Rahgooy** [comment]
@danyaljj I did `compile` and `format` it, but it seems that it did not fix the problem.

**Comment by danyaljj** [comment]
I restarted the CI; now it's happy ...

**Comment by danyaljj** [comment]
Could you move your sample test files into `test/resources`, instead of `main/resources`? 
Also, add a readme file `README.md` adding a preliminary explanation about the task? It doesn't have to be detailed.

Otherwise all look good.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/Dictionaries.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+object Dictionaries {
+
+  val prepositions = HashSet(
+    "about", "above", "across", "after", "against", "along", "among", "around", "as", "at", "before",
+    "behind", "beneath", "beside", "between", "by", "down", "during", "for", "from", "in", "inside",
+    "into", "like", "of", "off", "on", "onto", "over", "round", "through", "to", "towards", "with"
+  )
+
+  def isPreposition(word: String): Boolean = {
+    val w = if (word == null) "" else word.toLowerCase.trim
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/Dictionaries.scala]: Unlike Java, in Scala it is always recommended to use `Option`s. 
Here you can have:
```scala 
def isPreposition(wordOpt: Option[String]): Boolean = {
   wordOpt match { 
      case Some(word) => prepositions.contains(word.toLowerCase.tim)
      case None => false 
   }   
} 
```

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala [hunk 2]
```diff
@@ -6,12 +6,14 @@
   */
 package edu.illinois.cs.cogcomp.saulexamples.nlp.SpatialRoleLabeling
 
-import edu.illinois.cs.cogcomp.core.datastructures.ViewNames
+import java.lang.Boolean
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala]: sort the imports.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala [hunk 3]
```diff
@@ -8,15 +8,195 @@ package edu.illinois.cs.cogcomp.saulexamples.nlp.SpatialRoleLabeling
...
+  }
+
+  def getHeadwordId(t: HasSpan, ta: TextAnnotation, offset: IntPair): Int = {
+
+    if (tagIsNullOrOutOfSentence(t, offset))
+      return -1
+
+    val start = t.getStart().intValue() - offset.getFirst
+    val startTokenId = ta.getTokenIdFromCharacterOffset(start)
+    var headwordId = startTokenId
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala]: highly recommended not to use mutables ...

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsDataModel.scala [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      getFeature(x.getTrajector.getFirstConstituent, fex)
+  }
+  val JF2_10 = property(relations) {
+    x: RobertsRelation =>
+      {
+        val tokens = x.getTextAnnotation.getView(ViewNames.TOKENS)
+          .getConstituentsCoveringSpan(x.getFirstArg.getSpan.getFirst, x.getLastArg.getSpan.getSecond)
+          .asScala.toList
+        val list = ListBuffer[String]()
+        var spIncluded = false
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsDataModel.scala]: same here

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala (hunk 3 above)
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala]: OK, I'll change them. Thanks

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala [hunk 5]
```diff
@@ -199,16 +194,16 @@ object PopulateSpRLDataModel extends Logging {
...
       val headId = CollinsHeadFinder.getInstance.getHeadWordPosition(parsePhrase)
       val head = ta.getView(ViewNames.TOKENS).asInstanceOf[TokenLabelView].getConstituentAtToken(headId)
       if (constituents.exists(x => x.getSpan == head.getSpan) && isArgCandidate(head))
         return headId
 
-      val candiates = constituents.filter(c => isArgCandidate(c))
-      if (candiates.size > 0) {
-        val lastId = candiates.last.getStartSpan
+      val candidates = constituents.filter(c => isArgCandidate(c))
+      if (candidates.size > 0) {
```
Comment (danyaljj) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/PopulateSpRLDataModel.scala]: `candidates.nonEmpty`?

File: saul-examples/src/test/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsDataModelReaderTests.scala [hunk 6]
```diff
@@ -40,7 +40,7 @@ class RobertsDataModelReaderTests extends FlatSpec with Matchers {
   "Roberts Data Model Features" should "be correct for examples of the paper." in {
     val examples = sentences()
       .filter(x => x.getSentence.getSentenceConstituent.getTextAnnotation.getId.contains("example.xml"))
-      .map(x=> x.getSentence).toList
+      .map(x => x.getSentence).toList
```
Comment (danyaljj) [inline saul-examples/src/test/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsDataModelReaderTests.scala]: `.map(_.getSentence).toList`?

**Comment by danyaljj** [comment]
Since we changed our Semaphore account, it won't get triggered unless you close the PR and open a new one.

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsElement.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Sentence;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
+
+import java.util.List;
+
+/**
+ * Created by taher on 8/11/16.
+ */
+public class RobertsElement implements Comparable<RobertsElement> {
```
Comment (kordjamshidi) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsElement.java]: why not naming it just a triplet?

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsElementTypes.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package edu.illinois.cs.cogcomp.saulexamples.nlp.SpatialRoleLabeling;
+
+/**
+ * Created by taher on 8/11/16.
+ */
+public enum RobertsElementTypes {
+    INDICATOR, TRAJECTOR, LANDMARK
+}
```
Comment (kordjamshidi) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsElementTypes.java]: also naming here: RoleTypes?

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsRelation.java [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+ */
+public class RobertsRelation {
+    private final RobertsElement trajector;
+    private final RobertsElement spatialIndicator;
+    private final RobertsElement landmark;
+    private final RobertsRelationLabels label;
+    private final Sentence sentence;
+    private final List<RobertsElement> orderedArgs;
+    private final String relationId;
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/RobertsRelation.java]: `RobertsRelation` -> `SpRelation`?

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/SpRLDataModel.scala [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+      otherPrep match {
+        case Some(p) => p.toString
+        case None => ""
+      }
+  }
+
+  val JF2_12 = property(relations) {
+    x: SpRelation => ""
+    //      val view = x.getTextAnnotation.getView(ViewNames.SRL_VERB)
+    //      view match {
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/SpRLDataModel.scala]: what is this? why commented out?

Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/SpRLDataModel.scala]: This is for SRL features. They are disabled for now.

**Comment by kordjamshidi** [comment]
BTW, you need to make a new PR I guess for semaphore, right?

**Comment by Rahgooy** [comment]
Yes, I guess.
I'll close this, and make a fresh one

**Pull request closed without merging**